test/dependency-manager/validation.test.ts

```diff
@@ -0,0 +1,233 @@
+import { expect } from 'chai';
+import { flattenValidationErrors } from '../../src/common/utils/errors';
+import { ServiceConfigBuilder } from '../../src/dependency-manager/src';
+
+describe('validation (v1 spec)', () => {
```

Comment:
Can we make sure to throw an error on unknown values? `forbidUnknownValues`
https://github.com/typestack/class-validator#passing-options

Something as stupid as `parameter` vs `parameters`

Then add test cases to cover a typo at various nestings?

Reply:
Good idea. I'll make that the default of the `.validate()` method

Reply:
Cool - might cause issues with your valueFrom hack, but you could always define both with a comment
value_from
valueFrom